src/command_modules/azure-cli-interactive/azclishell/telemetry.py

```diff
@@ -27,19 +27,60 @@ def my_context(tel_client):
     tel_client.context.instrumentation_key = INSTRUMENTATION_KEY
 
 
+class TelThread(threading.Thread):
```

Comment:
Let's forget about the background thread idea for a minute and implement a one thread everytime, mech.